src/metabase/driver/util.clj

```diff
@@ -210,6 +210,19 @@
              :when  (driver/available? driver)]
          driver)))
 
+(defn semantic-version-gte
+  "Returns true if x is greater than or equal to y according to semantic versioning.
+   x and y are vectors of integers of the form `[major minor ...]`, where minor is optional.
+   Only the major and minor parts are compared.
+
+   Examples:
+   (semantic-version-gte [4 1] [4 1]) => true
+   (semantic-version-gte [4 0 1] [4 1]) => false
+   (semantic-version-gte [4 1] [4]) => true"
+  [x y]
+  (nat-int? (compare (vec (take 2 x))
+                     (vec (take 2 y)))))
```

Comment:
I'm a little worried that people are going to use this and not read the docstring and miss the part where it says 

> Only the major and minor parts are compared.

And accidentally shoot themselves in the foot. I would love if there was a way to prevent someone from accidentally doing something like

```clj
(defmethod driver/database-supports? ...
  ;; support for whateverFunction() was added in 4.0.1
  (semantic-version-gte database-version [4 0 1]))
```

I would be super happy. Code without gotchas or caveats is always preferrable. I think some options for how you could accomplish the make-it-hard-to-use-this-wrong goal would be

- Rename the function to something like `semantic-major-minor-version-gte` -- I think it's best to error on the side of too explicit rather than too vague when naming stuff. No one is going to use  `semantic-major-minor-version-gte` and accidentally think it also compares patch or whatever
- Just change the function to compare the whole thing so there are no caveats people have to know about. It would be a little bit of work to make it handle versions of different length but it's doable
- Make the expectation that you call it like `(semantic-version-gte actual-version expected-version)` explicit and assert that `expected-version` is exactly two parts

Reply:
For reference, Tamas fixed this with a new implementation https://github.com/metabase/metabase/pull/27176/files#diff-f0e79946f9f4750d54813a255d71aa5fa19218753231fc523369e5ab0469aaf8R203